### PR TITLE
community: Add document manager and mongo document manager

### PR DESCRIPTION
**Pull request by 2jimoo**
- **Description:** 
    - Add DocumentManager class, which is a nosql record manager. 
    - In order to use index and aindex in libs/langchain/langchain/indexes/_api.py, DocumentManager inherits RecordManager.
    - Also I added the MongoDB implementation of Document Manager too.
  - **Dependencies:** pymongo, motor

### CODE REVIEW

**Comment by 2jimoo** [comment]
@eyurtsev  May I ask if the PR is in progress?

File: libs/community/langchain_community/indexes/base.py [hunk 1]
```diff
@@ -170,3 +170,171 @@ async def adelete_keys(self, keys: Sequence[str]) -> None:
         Args:
             keys: A list of keys to delete.
         """
+
+
+class DocumentManager(RecordManager, ABC):
```
Comment (eyurtsev) [inline libs/community/langchain_community/indexes/base.py:175]: @2jimoo it seems that this has an identical interface to the record manager. Are you OK if we remove it?

File: libs/community/tests/integration_tests/indexes/docker-compose/mongodb.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+version: '3.1'
```
Comment (eyurtsev) [inline libs/community/tests/integration_tests/indexes/docker-compose/mongodb.yml:1]: Would you be able to move this to langchain/docker/docker-compose.yml?

**Review by eyurtsev — COMMENTED** [review]
This code looks great! Just a question about whether we can remove the DocumentManager abstract interface and I think we can merge in

File: libs/community/langchain_community/indexes/base.py (hunk 1 above)
Comment (2jimoo) [reply libs/community/langchain_community/indexes/base.py:175]: Yes, it's okay to remove it.

File: libs/community/tests/integration_tests/indexes/docker-compose/mongodb.yml (hunk 2 above)
Comment (2jimoo) [reply libs/community/tests/integration_tests/indexes/docker-compose/mongodb.yml:1]: Sure, I'll do that. Thanks for the review!

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
* Resolved merge conflicts
* updated auth information and port
* removed pytest async marker (since we use a plugin to automatically do that)

We can merge this as soon as unit tests pass

**Pull request merged**